server/controllers/workSpace.controller.ts

```diff
@@ -0,0 +1,141 @@
+          const workspaces = await this.workspaceService.getUserWorkspaces(
+            userId,
+          );
+          res.json(workspaces);
+        },
+      )
+      .post(
+        /**
+         * @openapi
+         * /sharkio/workspace:
+         *   post:
+         *     tags:
+         *      - workspace
+         *     description: Create a workspace
+         *     requestBody:
+         *        description: Create a workspace
+         *        content:
+         *          application/json:
+         */
+        async (req: Request, res: Response) => {
+          const { newWorkSpaceName } = req.body;
+          const userId = res.locals.auth.user.id;
+          const newWorkspace = await this.workspaceService.createWorkspace(
+            newWorkSpaceName,
+            userId,
+          );
+          res.json(newWorkspace);
+        },
+      );
+
+    router
+      .route("/:workspaceId")
+      .get(
+        /**
+         * @openapi
+         * /sharkio/workspace:
+         *   get:
+         *     tags:
+         *      - workspace
+         *     description: get requested workspace
+         *     responses:
+         *       200:
+         *         description: Returns requested workspace
+         *       500:
+         *         description: Server error
+         */
+        async (req: Request, res: Response) => {
+          const userId = res.locals.auth.user.id;
+          const { workspaceId } = req.params;
+          const workspace = await this.workspaceService.getWorkspace(
+            userId,
+            workspaceId,
+          );
+          res.json(workspace);
+        },
+      )
+      .delete(
+        /**
+         * @openapi
+         * /sharkio/workspace/{workspaceId}:
+         *   delete:
+         *     tags:
+         *      - workspace
+         *     description: delete requested workspace
+         *     responses:
+         *       200:
+         *         description: Returns requested workspace
+         *       500:
+         *         description: Server error
+         */
+        async (req: Request, res: Response) => {
+          const userId = res.locals.auth.user.id;
+          const { workspaceId } = req.params;
+          await this.workspaceService.deleteWorkspace(userId, workspaceId);
+          res.json({ success: true });
+        },
+      )
+      .put(
+        /**
+         * @openapi
+         * /sharkio/workspace/{workspaceId}:
+         *   put:
+         *     tags:
+         *      - workspace
+         *     description: change workspace name
+         *     responses:
+         *       200:
+         *         description: Returns requested workspace
+         *       500:
+         *         description: Server error
+         */
+        async (req: Request, res: Response) => {
+          const userId = res.locals.auth.user.id;
+          const { workspaceId } = req.params;
+          const { newWorkspaceName } = req.body;
```

Comment:
This could be just `name`

Reply:
You can have the same name twice so i will use the id.
But the user id is not needed

Reply:
I'm not sure i got your comment